### ISSUE TITLE
Limit how a user becomes an admin

### ISSUE THREAD

**Issue by humphd** [Seneca-CDOT/telescope#2153]
We have the concept of an `admin` role in the User and Auth services, in the front-end, etc.  In Firebase it's a boolean, `isAdmin: true|false`.  We need to tighten how a user becomes an admin.

You shouldn't be able to POST that data to the Users service (we should ignore it).   The current system is too loose.

**Comment by humphd** [comment]
```
POST /:id   - a user who owns id, or admin, or service (exclude isAdmin)
PUT /:id   - a user who owns id, or admin, or service (exclude isAdmin)
PUT /:id/admin - only an admin can do this?
```

**Comment by humphd** [comment]
Based on the changes I just made in #2344, we can probably do this by insisting that only an admin (super user via env, or existing admin) can **update** an existing user to an Admin in the db.  The work in #2153 is heading in this direction, we just need to finish it.

**Comment by humphd** [comment]
@chrispinkney and @manekenpix started this work in https://github.com/Seneca-CDOT/telescope/pull/2239, and we can likely build on the initial PR there to get this done, once the users backend is in place.

**Comment by humphd** [comment]
In #3030 I'm suggesting that we use GitHub OAuth with org/team membership to define our admins.  Closing this in favour of that.
